<launch>
  <!--原雷达的ip地址为 192.168.1.200   修改雷达的 target ip为 192.168.80.2-->
  <arg name="device_ip" default="192.168.80.2" />   
  <arg name="msop_port" default="2368" />
  <arg name="difop_port" default="2369" />
  <arg name="return_mode" default="1" />
  <arg name="degree_mode" value="1"/>
  <arg name="time_synchronization" default="true" />

  <node pkg="lslidar_c32_driver" type="lslidar_c32_driver_node" name="lslidar_c32_driver_node" output="screen">
    <param name="device_ip" value="$(arg device_ip)" />
    <param name="msop_port" value="$(arg msop_port)" />
    <param name="difop_port" value="$(arg difop_port)"/>
    <!--lego loam frame id 为 velodyne -->
    <!--<param name="frame_id" value="laser_link"/>-->
    <param name="frame_id" value="velodyne"/>
    <param name="add_multicast" value="false"/>
    <param name="group_ip" value="224.1.1.2"/>
    <param name="rpm" value="600"/>
    <param name="return_mode" value="$(arg return_mode)"/>
    <param name="degree_mode" value="$(arg degree_mode)"/>
    <param name="time_synchronization" value="$(arg time_synchronization)"/>
  </node>

  <node pkg="lslidar_c32_decoder" type="lslidar_c32_decoder_node" name="lslidar_c32_decoder_node" output="screen">
    <param name="min_range" value="0.15"/>
    <param name="max_range" value="150.0"/>
    <param name="degree_mode" value="$(arg degree_mode)"/>
    <param name="distance_unit" value="0.4"/>
    <param name="return_mode" value="$(arg return_mode)"/>
    <param name="config_vert" value="true"/>
    <param name="print_vert" value="false"/>
    <param name="scan_frame_id" value="laser_link"/>
    <param name="scan_num" value="15"/>
    <param name="publish_scan" value="true"/>
    <param name="time_synchronization" value="$(arg time_synchronization)"/>
    <!--lego loam 点云话题改为 /velodyne_points-->
    <remap from="lslidar_point_cloud" to="/velodyne_points" />
  </node>
  

  <!--node name="rviz" pkg="rviz" type="rviz" args="-d $(find lslidar_c32_decoder)/launch/lslidar_c32.rviz" output="screen"/-->

</launch>
